mlpipeline/custom/user_components.py

```diff
@@ -0,0 +1,43 @@
+import logging
+import numpy as np
+from mlpipeline.pipeline import Component
+
+##
+L = logging.getLogger(__name__)
+##
+
+
+class ImagePreprocessor(Component):
```

Comment:
Using inheritance does not seem to bring any benefits here, it couples the individual models with the generalComponent class and it's not the best separation of concerns. It could be replaced by a mapping of `component_name → callable` and the executor could look up the particular callable and pass in inputs.

Reply:
didn't really understand the reasoning behind, likely because I don't know actual models behaviour (hence separation). But note taken